chainer/datasets/tox21.py

```diff
@@ -0,0 +1,150 @@
+import numpy
+
+try:
+    from rdkit import Chem
+    from rdkit.Chem import rdMolDescriptors
+    _available = True
+except ImportError:
+    _available = False
+
+
+def check_available():
+    """Checks availability of Tox21 dataset.
+
+    This function checks the availability of Tox21 dataset
+    in user's environment.
+    Specifically, we use `RDKit <https://github.com/rdkit/rdkit>`_
+    to extract features and labels from raw files, whose format are
+    `SDF <https://en.wikipedia.org/wiki/Chemical_table_file#SDF>`_.
+    So, it returns ``True`` if Chainer successfully imports the
+    RDKit module.
+
+    Returns:
+        ``True`` is Tox21 dataset is available.
+        Otherwise ``False``.
+
+    """
+    if not _available:
+        warnings.warn('rdkit is not install on your environment '
+                      'Please install it to use tox21 dataset.\n'
+                      'See the official document for installation.'
+                      'http://www.rdkit.org/docs/Install.html')
+    return _available
+
+
+config = {
+    'train': {
+        'url': 'https://tripod.nih.gov/tox21/challenge/download?'
+        'id=tox21_10k_data_allsdf',
+        'filename': 'tox21_10k_data_all.sdf'
+    },
+    'val': {
+        'url': 'https://tripod.nih.gov/tox21/challenge/download?'
+        'id=tox21_10k_challenge_testsdf',
+        'filename': 'tox21_10k_challenge_test.sdf'
+    },
+    'test': {
+        'url': 'https://tripod.nih.gov/tox21/challenge/download?'
+        'id=tox21_10k_challenge_scoresdf',
+        'filename': 'tox21_10k_challenge_score.sdf'
+    }
+}
+
+root = 'pfnet/chainer/tox21'
+
+label_names = ['NR-AR', 'NR-AR-LBD', 'NR-AhR', 'NR-Aromatase', 'NR-ER',
+               'NR-ER-LBD', 'NR-PPAR-gamma', 'SR-ARE', 'SR-ATAD5',
+               'SR-HSE', 'SR-MMP', 'SR-p53']
+
+
+def _ECFP(mol_supplier, label_names, radius=2):
+    descriptors = []
+    labels = []
+    for mol in mol_supplier:
+        if mol is None:
+            continue
+        label = []
+        for task in label_names:
+            if mol.HasProp(task):
+                label.append(int(mol.GetProp(task)))
+            else:
+                label.append(-1)
+        try:
+            fp = rdMolDescriptors.GetMorganFingerprintAsBitVect(mol, radius)
+        except Exception:
+            continue
+        descriptors.append(fp)
+        labels.append(label)
+    descriptors = numpy.array(descriptors, dtype=numpy.float32)
+    labels = numpy.array(labels, dtype=numpy.int32)
+    return descriptors, labels
+
+
+def _creator(cached_file_path, sdffile, url):
+    download_file_path = download.cached_download(url)
+
+    with zipfile.ZipFile(download_file_path, 'r') as z:
+        z.extract(sdffile)
+    mol_supplier = Chem.SDMolSupplier(sdffile)
+    shutil.move(sdffile, cached_file_path)
+    return mol_supplier
+
+
+def _get_tox21(config_name, preprocessor, with_label=True):
+    basename = config_name
+    global config
```

Comment:
You don't need this line, because you're not rewriting `config` reference itself.